src/resolvers/hostSystemProfile/hostSystemProfile.integration.ts

```diff
@@ -378,5 +378,219 @@ describe('host system profile', function () {
+            expect(status).toEqual(200);
+            data.hostSystemProfile.operating_system.meta.should.eql({total: 3, count: 3});
+            data.hostSystemProfile.operating_system.data.should.eql([C33, R11, R12]);
+        });
+
+        describe('pagination', function () {
+            test('limit', async () => {
+                await createHosts(...hosts);
+
+                const { data, status } = await runQuery(QUERY, {
+                    limit: 2
+                }, getContext().headers);
+                expect(status).toEqual(200);
+
+                data.hostSystemProfile.operating_system.meta.should.eql({total: 3, count: 2});
+                data.hostSystemProfile.operating_system.data.should.eql([C33, R11]);
+            });
+
+            test('limit + offset', async () => {
+                await createHosts(...hosts);
+
+                const { data, status } = await runQuery(QUERY, {
+                    limit: 2,
+                    offset: 2
+                }, getContext().headers);
+                expect(status).toEqual(200);
+
+                data.hostSystemProfile.operating_system.meta.should.eql({total: 3, count: 1});
+                data.hostSystemProfile.operating_system.data.should.eql([R12]);
+            });
+        });
+
+        describe('ordering', function () {
+            test('ordering value DESC', async () => {
+                await createHosts(...hosts);
+
+                const { data, status } = await runQuery(QUERY, {
+                    order_by: 'value',
+                    order_how: 'DESC'
+                }, getContext().headers);
+                expect(status).toEqual(200);
+
+                data.hostSystemProfile.operating_system.meta.should.eql({total: 3, count: 3});
+                data.hostSystemProfile.operating_system.data.should.eql([R12, R11, C33]);
+            });
+        });
+
+        test('host filter', async () => {
+            await createHosts(...hosts);
+
+            const { data, status } = await runQuery(QUERY, {
+                hostFilter: {
+                    display_name: {
+                        matches: 'foo04'
+                    }
+                }
+            }, getContext().headers);
+            expect(status).toEqual(200);
+
+            data.hostSystemProfile.operating_system.meta.should.eql({total: 1, count: 1});
+            data.hostSystemProfile.operating_system.data.should.eql([C33]);
+        });
+
+        test('Query hosts with partial operating system', async () => {
+            await createHosts(...hosts);
+            //All hosts with partial OS versions should effectively be ignored
+            await createHosts(...partialOSHosts);
+
+            const { data, status } = await runQuery(QUERY, {}, getContext().headers);
+            expect(status).toEqual(200);
+            data.hostSystemProfile.operating_system.meta.should.eql({total: 4, count: 3});
```

Comment:
`{total: 4, count: 3}` - is the total including the null OS?

Reply:
Yikes, looks like it is, I thought it was because one of the buckets had two hosts in it, but that's not how it works. I re-assessed the query and now elastic search handles the exclusion so it should be faster, simpler, and more consistent.